cmd/gateway/setup.go

```diff
@@ -120,3 +123,22 @@ func MustValidateArguments(flagset *flag.FlagSet, validators ...ValidatorContext
 		os.Exit(1)
 	}
 }
+
+func getBuildInfo() (commitHash string, commitTime string, dirtyBuild bool) {
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		return "unknown", "unknown", true
+	}
+	for _, kv := range info.Settings {
```

Comment:
(sorry, missed this in the first review)

is there are a chance any of those settings are not present in `info.Settings` ? in that case, are we're ok with reporting empty strings? perhaps "unknown" would work better?

do we want a unit test to test this logic? 

wonder what @kate-osborn thinks

Reply:
ah, good catch! I've updated it

Reply:
I think we should add unit tests wherever possible.

Reply:
As I understand it that information is not available when running the tests. We could manually build the binary and then check that the info is present, but maybe it doesn't belong in a unit test?

Reply:
one approach is like below:
```go
type buildInfoFunc func() (info *debug.BuildInfo, ok bool)

var buildInfo buildInfoFunc = debug.ReadBuildInfo

func getBuildInfo() (commitHash string, commitTime string, dirtyBuild string) {
	commitHash = "unknown"
	commitTime = "unknown"
	dirtyBuild = "unknown"

	info, ok := buildInfo()
	if !ok {
		return
	}
	for _, kv := range info.Settings {
		switch kv.Key {
		case "vcs.revision":
			commitHash = kv.Value
		case "vcs.time":
			commitTime = kv.Value
		case "vcs.modified":
			dirtyBuild = kv.Value
		}
	}

	return
}
```

this way it will be possible to write and use a stub of buildInfo to use in the unit test

Reply:
I was thinking about this, and if we can't test the actual values, there's no logic (that we implemented) to test in this function as there's just a loop reading values from strings. Right?

Reply:
I see a few cases here:
- normal case: buildInfo() returns OK with values populated in info.Settings 
- edge case: buildInfo() returns not OK
- edge case: buildInfo() returns  OK without values got populated info.Settings 

in all those cases, we can ensure that the getBuildInfo() returns the expected values. There is some amount of logic, since we have `if` and  a loop.